airflow/providers/databricks/hooks/databricks.py

```diff
@@ -404,6 +483,35 @@ def start_cluster(self, json: dict) -> None:
         """
         self._do_api_call(START_CLUSTER_ENDPOINT, json)
 
+    def activate_cluster(self, json: dict, polling: int, timeout: int = -1) -> None:
+        """
+        Start the cluster, and wait for it to be ready.
+
+        :param json: json dictionary containing cluster specification.
+        :param polling: polling interval in seconds.
+        :param timeout: timeout in seconds. -1 means no timeout.
+        """
+        api_called = False
+        elapsed_time = 0
+
+        while (timeout == -1) or (elapsed_time <= timeout):
+            run_state = self.get_cluster_state(cluster_id)
+
+            if run_state.is_running:
+                return
+            elif run_state.is_terminal:
+                if not api_called:
+                    self.start_cluster({'cluster_id': cluster_id})
+                    api_called = True
+                else:
+                    raise AirflowException(
+                        f"Cluster {cluster_id} start failed with '{run_state.state}' state: {run_state.state_message}"
+                    )
+
+            # wait for cluster to start
+            time.sleep(polling)
+            elapsed_time += polling
```

Comment:
```suggestion
        while True:
            run_state = self.get_cluster_state(cluster_id)

            if run_state.is_running:
                return
            elif run_state.is_terminal:
                if not api_called:
                    self.start_cluster({'cluster_id': cluster_id})
                    api_called = True
                else:
                    raise AirflowException(
                        f"Cluster {cluster_id} start failed with '{run_state.state}' state: {run_state.state_message}"
                    )

            # wait for cluster to start
            time.sleep(polling)

            elapsed_time += polling
            if timeout and elapsed_time <= timeout:
                break
            
```

Reply:
Maybe we can raise `AirflowException`, I will change it to raise exception :)